source/physics/src/GateJPETParaPlusPromptDecayModel.cc

```diff
@@ -0,0 +1,97 @@
+/**
+ *  @copyright Copyright 2018 The J-PET Gate Authors. All rights reserved.
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  @file GateJPETParaPlusPromptDecayModel.cc
+ */
+
+#include "GateJPETParaPlusPromptDecayModel.hh"
+#include "TLorentzVector.h"
+#include "G4Electron.hh"
+#include "G4PhysicalConstants.hh"
+#include "GateConstants.hh"
+#include "G4SystemOfUnits.hh"
+#include <cmath>
+
+GateJPETParaPlusPromptDecayModel* GateJPETParaPlusPromptDecayModel::ptrJPETParaPlusPromptDecayModel=0;
+
+GateJPETParaPlusPromptDecayModel::GateJPETParaPlusPromptDecayModel()
+{
+ G4cout <<"GateJPETParaPlusPromptDecayModel initialization.\n";
+ SetParticlesNumber(3);
+ GateJPETSourceManager::GetInstance()->AddGammaSourceModel(this);
+ ptrRandomGenerator = new TRandom3( 0 );
+}
+
+GateJPETParaPlusPromptDecayModel::~GateJPETParaPlusPromptDecayModel()
+{
+}
+
+void GateJPETParaPlusPromptDecayModel::GetGammaParticles(std::vector<G4PrimaryParticle*>& particles)
+{
+ AddGammaFromDeexcitation( particles );
+ AddGammasFromParaPositronium( particles );
+}
+
+G4String GateJPETParaPlusPromptDecayModel::GetModelName()
+{
+ return "pPsAndPropmtJPET";
+}
+
+GateJPETParaPlusPromptDecayModel* GateJPETParaPlusPromptDecayModel::GetInstance()
+{
+ if(!ptrJPETParaPlusPromptDecayModel)
+  ptrJPETParaPlusPromptDecayModel = new GateJPETParaPlusPromptDecayModel;
+ return ptrJPETParaPlusPromptDecayModel;
+}
+
+
+void GateJPETParaPlusPromptDecayModel::AddGammaFromDeexcitation( std::vector<G4PrimaryParticle*>& particles )
+{
+ G4ThreeVector momementum_direction = GetRandomVectorOnSphere();
+ G4double kinetic_energy = 1.157 * MeV; //Add here command
```

Comment:
this energy can be set by some method?
Then it would be useful for different sources e.g. Scandium or Na

Reply:
if not it should be const anyway